File: packages/types/src/category.ts
export type CategoryType = "income" | "expense";

export type CategoryBucket = "necessity" | "desire" | "save_invest";

export interface Category {
  id: string;
  name: string;
  type: CategoryType;
  bucket?: CategoryBucket;
  icon: string;
  color: string;
  isDefault: boolean;
  createdAt: string;
}

export interface CreateCategoryInput {
  name: string;
  type: CategoryType;
  bucket?: CategoryBucket;
  icon?: string;
  color?: string;
}

File: apps/api/app/schemas/category.py
from pydantic import BaseModel


class CreateCategoryInput(BaseModel):
    name: str
    type: str  # income | expense
    bucket: str | None = None
    icon: str | None = "tag"
    color: str | None = "#64748b"


class UpdateCategoryInput(BaseModel):
    name: str | None = None
    type: str | None = None
    bucket: str | None = None
    icon: str | None = None
    color: str | None = None


class CategoryResponse(BaseModel):
    id: str
    name: str
    type: str
    bucket: str | None = None
    icon: str
    color: str
    isDefault: bool
    createdAt: str
